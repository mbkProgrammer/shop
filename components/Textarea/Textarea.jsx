import { useTheme } from '@emotion/react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

const Textarea = ({ placeholder, size }) => {
  const theme = useTheme();

  const Textarea = styled.textarea`
  height: 80px;
  margin: 5px;
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  border: 0.5px solid ${theme.colors.primary};
  border-radius: 5px;
  width: 445px;
  transition: 0.3s;
  background: ${theme.colors.secondary};
  &:focus {
    outline: none;
    border: 2px solid ${theme.colors.primary} ;
  }
  &:hover {
    background-image: linear-gradient(${theme.colors.background} 0 0);
  }
  &.input--big{
    width: 445px;
    max-width: 445px;
  }
  &.input--small{
    max-width: 140px;
  }
  &.input--medium{
    max-width: 216px;
  }
  @media (max-width: 480px){
    &.input--big{
      width: 347px;
    }
  @media (max-width: 350px){
    &.input--big{
      width: 307px;
    }
  }
  `;
  return (
    <Textarea className={`input--${size}`} placeholder={placeholder} />
  );
};

Textarea.propTypes = {
  placeholder: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'big']),
};

Textarea.defaultProps = {
  placeholder: 'placeholder',
  size: 'medium',
};

export default Textarea;
