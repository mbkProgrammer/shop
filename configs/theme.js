const theme = {
  colors: {
    primary: '#56B280',
    secondary: '#F5FBF8',
    text: '#222322',
    background: '#EAF6F0',
  },
  typography: {
    body: {
      fontWeight: '300',
      fontSize: '1rem',
    },
    h1: {
      fontWeight: '700',
      fontSize: '2.4rem',
    },
    h2: {
      fontWeight: '700',
      fontSize: '2.2rem',
    },
    h3: {
      fontWeight: '700',
      fontSize: '2rem',
    },
    h4: {
      fontWeight: '600',
      fontSize: '1.8rem',
    },
    h5: {
      fontWeight: '600',
      fontSize: '1.6rem',
    },
    h6: {
      fontWeight: '500',
      fontSize: '1.4rem',
    },
  },
};

export default theme;
