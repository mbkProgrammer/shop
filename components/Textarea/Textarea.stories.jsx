import React from 'react';
import Textarea from './Textarea';

export default { title: 'components/Textarea' };
export const TextareaSmall = () => (<Textarea size="small" placeholder="Small Textarea" />);
export const TextareaMedium = () => (<Textarea size="medium" placeholder="Small Textarea" />);
export const TextareaBig = () => (<Textarea size="big" placeholder="Small Textarea" />);
