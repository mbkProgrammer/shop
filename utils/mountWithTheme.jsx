import { ThemeProvider } from '@emotion/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../configs/Store';
import theme from '../configs/theme';

const mountWithTheme = (children) => renderer.create(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </Provider>,
);

export default mountWithTheme;
