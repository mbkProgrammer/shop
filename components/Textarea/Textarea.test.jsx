import { ThemeProvider } from '@emotion/react';
import React from 'react';
import renderer from 'react-test-renderer';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import Textarea from './Textarea';
import theme from '../../configs/theme';
import mountWithTheme from '../../utils/mountWithTheme';

test('Textarea smaple', () => {
  const component = mountWithTheme(
    <Textarea placeholder="Textarea test" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Textarea smaple type password', () => {
  const component = mountWithTheme(
    <Textarea placeholder="Textarea test" type="password" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Textarea smaple size big', () => {
  const component = mountWithTheme(
    <Textarea placeholder="Textarea test" size="big" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Textarea smaple onChange', () => {
  let variable = '';
  const { getByTestId } = render(
    <ThemeProvider theme={theme}>
      <Textarea varaint="contained" onChange={(e) => { variable += e.target.value; }} />
    </ThemeProvider>,
  );

  const textarea = screen.getByTestId('Textarea--test');
  fireEvent.change(textarea, { target: { value: 'a' } });
  expect(variable).toBe('a');
});
