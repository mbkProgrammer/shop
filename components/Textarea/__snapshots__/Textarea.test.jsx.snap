// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea smaple 1`] = `
<textarea
  className="input--medium css-catwbv"
  data-testid="Textarea--test"
  placeholder="Textarea test"
/>
`;

exports[`Textarea smaple size big 1`] = `
<textarea
  className="input--big css-catwbv"
  data-testid="Textarea--test"
  placeholder="Textarea test"
/>
`;

exports[`Textarea smaple type password 1`] = `
<textarea
  className="input--medium css-catwbv"
  data-testid="Textarea--test"
  placeholder="Textarea test"
/>
`;
